detect_s3.go

```diff
@@ -59,3 +61,13 @@ func (d *S3Detector) detectVhostStyle(region, bucket string, parts []string) (st
 
 	return "s3::" + url.String(), true, nil
 }
+
+func (d *S3Detector) detectNewVhostStyle(region, bucket string, parts []string) (string, bool, error) {
```

Comment:
NIT: I wonder what the new, new name will be if AWS adds a new 6 parts vhost-style url